_source/_posts/2021-08-06-webforms-blazor-migration.md

```diff
@@ -0,0 +1,839 @@
+        <Authorizing>
+            <p>Authorizing please wait...</p>
+        </Authorizing>
+        <NotAuthorized>
+            <h1>Sorry</h1>
+            <p>You're not authorized to see this page. You need to log in.</p>
+        </NotAuthorized>
+        </AuthorizeRouteView>
+    </Found>
+    <NotFound>    
+        <p>Sorry, there's nothing at this address.</p>    
+    </NotFound>
+</Router>
+```
+
+The `AuthorizeRouteView` component ensures that you can see only the pages that the user is authorized to visit.
+
+After this, open the `CaloriesTracking.razor` component and add `@attribute [Authorize]` at the top of the code so that only authorized users can view it.
+
+Finally, you need a way to authenticate users. I will create a new component, `AccountControl` in the `Shared` folder and put the following code:
+
+```html
+@inject NavigationManager NavigationManager
+
+<AuthorizeView>
+    <Authorized>
+        <span>Hello, @context.User.Identity.Name!</span>
+        <a href="logout">Log out</a>
+    </Authorized>
+    <NotAuthorized>
+        <a href="login?redirectUri=@(NavigationManager.Uri)">Log in</a>
+    </NotAuthorized>
+</AuthorizeView>
+```
+
+The `AuthorizeView` view will either display a *Log out* link for the user or a *Log in* link for authentication. The `AuthorizeView` component exposes a `context` variable, which I use to get the information about the logged-in user.
+
+Put the component in the `MainLayout.razor` file above the *About* link and run the app. When you click the **Calories Tracker** you will see a message that you don't have access to this page:
+
+{% img blog/webforms-blazor-migration/image8.png alt:"" width:"800" %}{: .center-image }
+
+Click the **Log in** link, and the app will redirect you to Okta for authentication. When you log in, Okta sends you back to the app, where you'll be able to access the page that only authenticated users can see:
+
+{% img blog/webforms-blazor-migration/image9.png alt:"" width:"800" %}{: .center-image }
+
+That's it! You have now finished migrating a WebForms project to Blazor and secured it with Okta.
+
+Give Blazor a try. Get the [complete source code of the project from GitHub](https://github.com/oktadev/Okta.CaloriesTracker), and let us know what you think in the comments below.
+
+- [Goodbye Javascript! Build an Authenticated Web App in C# with Blazor + ASP.NET Core 3.0](https://developer.okta.com/blog/2019/10/16/csharp-blazor-authentication)
+- [How to Build Securely with Blazor WebAssembly (WASM)](https://developer.okta.com/blog/2020/09/30/blazor-webassembly-wasm-dotnetcore)
+- [ASP.NET Core 3.0 MVC Secure Authentication](https://developer.okta.com/blog/2019/11/15/aspnet-core-3-mvc-secure-authentication)
+
+Make sure to follow us on [Twitter](https://twitter.com/oktadev) and subscribe to our [YouTube Channel](https://youtube.com/c/oktadev) and check out our new [Twitch](https://www.twitch.tv/oktadev) live stream so that you never miss any awesome content!
```

Comment:
```suggestion
Make sure to follow us on [Twitter](https://twitter.com/oktadev),  subscribe to our [YouTube Channel](https://youtube.com/c/oktadev), and check out our new [Twitch](https://www.twitch.tv/oktadev) live stream so that you never miss any awesome content!
```
"and" reduction in a series...